develop/learning-paths/migration/articles/01-intro/01-from-a-monolith-to-a-modular-architecture.markdown

```diff
@@ -0,0 +1,124 @@
+# From a Monolith to a Modular Architecture
+
+Since Liferay Portal's beginning it's been an amazing application full of all
+kinds of capabilities. It's had powerful built-in apps and extensible frameworks
+that deploy on all kinds of devices. These apps leverage Portal's powerful
+frameworks (e.g., asset and comment frameworks) and in some cases the apps
+leverage each other (e.g., the Blogs and Wiki apps leverage the Documents and
+Media Library). And undoubtedly you've leveraged Liferay Portal's apps and
+frameworks via their APIs. Yes, Liferay Portal has traditionally provided so
+many things to so many users and developers--and that's been a very good thing.
+
+But as Liferay Portal's grown, it has become a monolith of sorts. Its repository
+code base was huge and it's memory footprint was much bigger than what we'd
+prefer. Considering all that Portal had been providing and how it's evolved, its
+size isn't surprising, but we knew we had to do better for Liferay 7. So before
+we started designing Liferay 7, we took a step back to consider how to provide
+even better capabilities while making Liferay Portal, and its features and
+frameworks, lean, agile, and independent. To start, we took a hard look at key
+capabilities to improve.
+
```

Comment:
I think this reveals much too much of Liferay's baggage. It's better to emphasize the benefits of modularity for the *future* rather than focus on problems of the past. Otherwise, we unintentionally impugn our past products and the customers who chose them. So instead of talking about our huge codebase, big memory footprint, and problems we had with releaseability, it's better to say we wanted to streamline our codebase, make Liferay more tunable, and split parts of Liferay into independent chunks. This way, we say we're making improvements to the same things without saying we had all kinds of problems.